### DISCUSSION TITLE
AnimatedBox predefined component

### DISCUSSION THREAD

**Discussion by hamdij0maa** [Shopify/restyle#230] [Ideas]
Can we add this : 

import {Theme} from './theme';

const Box = createBox<Theme>();

export default Box;

type AnimatedViewProps = Animated.AnimateProps<
  ViewProps & {children?: ReactNode}
>;

type Props = BoxProps<Theme> & AnimatedViewProps;

export const AnimatedBox = createRestyleComponent<Props, Theme>(
  [spacing, backgroundColor, spacing, border, shadow],
  Animated.View,
);
``

**Comment by Johan-dutoit** [comment]
It would be super if this worked, however it all goes wonky with hot reloading.

```typescript
type BoxProps = ComponentProps<typeof Box>;
type AnimatedViewProps = AnimatedProps<ViewProps & {children?: ReactNode}>;
type Props = BoxProps & AnimatedViewProps;
export const AnimatedBox = createBox<Theme, Props>(Animated.View);
```


I've ended up going with something like this

```typescript
type BoxProps = ComponentProps<typeof Box>;
type AnimatedViewProps = AnimatedProps<ViewProps & {children?: ReactNode}>;
type Props = BoxProps & AnimatedViewProps;

export const AnimatedBox = createRestyleComponent<
  BoxProps & Omit<Props, keyof BoxProps>,
  Theme
>(
  boxRestyleFunctions as RestyleFunctionContainer<BoxProps, Theme>[],
  Animated.View,
);
```

**Reply by nicktiberi** [reply]
Alternatively you can wrap the other way around, by wrapping the `Box` using reanimated:

```
export const AnimatedBox = Animated.createAnimatedComponent(Box);
```
